### PR TITLE
Feature/add x unit

### PR DESCRIPTION
**Pull request by Blackolf931**

### CODE REVIEW

File: TSP/BLL/Services/EmployeeService.cs [hunk 1]
```diff
@@ -44,6 +44,10 @@ public async Task<Employee> GetEmployeeByIdAsync(int id)
         {
             var employee = await _repository.GetByIdAsync(id);
             var mappedEmployee = _mapper.Map<Employee>(employee);
+            if(employee is null)
```
Comment (nantonov) [inline TSP/BLL/Services/EmployeeService.cs:47]: Should be checked after getting the employee

File: TSP/BLL/Services/RetirePeopleSetadditionalInfo.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace BLL.Services
 {
-    class RetirePeopleSetadditionalInfo : IStrategy
+    public class RetirePeopleSetadditionalInfo : IStrategy
```
Comment (nantonov) [inline TSP/BLL/Services/RetirePeopleSetadditionalInfo.cs:5]: The name of these classes should end with "*Startegy".

File: TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+using BLL.Services;
+using DAL.Entities;
+using DAL.Interfaces;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace TestsServicesTSP.Moq
```
Comment (nantonov) [inline TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs:13]: Should be named as a project which you test, ex. "{ProjectName.UnitTests}"

File: TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    public class EmployeeServiceTests
+    {
+        private readonly EmployeeService _sut;
+        private readonly Mock<IEmployeeRepository> _employeeRepoMock = new();
+        private readonly IEnumerable<IStrategy> strategies = new List<IStrategy> { new MiddlePeopleSetAdditionalInfo(), new RetirePeopleSetadditionalInfo(), new YoungPeopleSetAdditionalInfo() };
+
+
+        public EmployeeServiceTests()
+        {
+            var mockMapper = new MapperConfiguration(cfg =>
```
Comment (nantonov) [inline TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs:24]: Unnecessary

File: TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            {
+                Id = employeeId,
+                Name = "Test",
+                SecondName = "Test",
+                Patronomic = "Test",
+                Age = 25,
+                Position = "Test",
+                OfficeId = 2
+            };
+            _employeeRepoMock.Setup(x => x.GetByIdAsync(19)).ReturnsAsync(employeeEntity);
```
Comment (nantonov) [inline TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs:47]: What is 19? Get rid of this

**Review by nantonov — CHANGES_REQUESTED** [review]

File: TSP/BLL/Services/EmployeeService.cs (hunk 1 above)
Comment (Blackolf931) [reply TSP/BLL/Services/EmployeeService.cs:47]: fix

File: TSP/BLL/Services/RetirePeopleSetadditionalInfo.cs (hunk 2 above)
Comment (Blackolf931) [reply TSP/BLL/Services/RetirePeopleSetadditionalInfo.cs:5]: fix

File: TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs (hunk 5 above)
Comment (Blackolf931) [reply TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs]: fix

File: TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs (hunk 3 above)
Comment (Blackolf931) [reply TSP/TestsServicesTSP.Moq/EmployeeServiceTests.cs:13]: fix

**Pull request merged**